app/models/reaction.rb

```diff
@@ -13,11 +14,12 @@ class Reaction < ApplicationRecord
 
   counter_culture :reactable,
                   column_name: proc { |model|
-                    model.points.positive? ? "positive_reactions_count" : "reactions_count"
+                    PUBLIC_CATEGORIES.include?(model.category) ? "public_reactions_count" : "reactions_count"
                   }
   counter_culture :user
 
-  scope :positive, -> { where("points > ?", 0) }
+  scope :positive, -> { where(category: PUBLIC_CATEGORIES) }
```

Comment:
Leaving this in here because we have one cache relying on this.

Reply:
You also don't need this since what is in the cache is the AR relation results rather than `.positive` scope

```
[30] pry(main)> Rails.cache.fetch("test-key") do
[30] pry(main)*   User.last.reactions.positive
[30] pry(main)* end
Cache read: test-key ({:namespace=>nil, :compress=>true, :compress_threshold=>1024, :expires_in=>3600, :race_condition_ttl=>nil})
Cache generate: test-key ({:namespace=>nil, :compress=>true, :compress_threshold=>1024, :expires_in=>3600, :race_condition_ttl=>nil})
  User Load (27.9ms)  SELECT  "users".* FROM "users" ORDER BY "users"."id" DESC LIMIT $1  [["LIMIT", 1]]
Cache write: test-key ({:namespace=>nil, :compress=>true, :compress_threshold=>1024, :expires_in=>3600, :race_condition_ttl=>nil})
  Reaction Load (15.1ms)  SELECT "reactions".* FROM "reactions" WHERE "reactions"."user_id" = $1 AND (points > 0)  [["user_id", 11]]
=> [#<Reaction:0x00007fe2c21ab448 id: 35, category: "unicorn", created_at: Mon, 18 May 2020 20:09:40 UTC +00:00, points: 1.0, reactable_id: 8, reactable_type: "Article", status: "valid", updated_at: Mon, 18 May 2020 20:09:40 UTC +00:00, user_id: 11>]
[31] pry(main)> Rails.cache.fetch("test-key").count
Cache read: test-key ({:namespace=>nil, :compress=>true, :compress_threshold=>1024, :expires_in=>3600, :race_condition_ttl=>nil})
   (5.4ms)  SELECT COUNT(*) FROM "reactions" WHERE "reactions"."user_id" = $1 AND (points > 0)  [["user_id", 11]]
=> 1
[32] pry(main)> Rails.cache.fetch("test-key")
Cache read: test-key ({:namespace=>nil, :compress=>true, :compress_threshold=>1024, :expires_in=>3600, :race_condition_ttl=>nil})
  Reaction Load (0.5ms)  SELECT "reactions".* FROM "reactions" WHERE "reactions"."user_id" = $1 AND (points > 0)  [["user_id", 11]]
=> [#<Reaction:0x00007fe2c1755598 id: 35, category: "unicorn", created_at: Mon, 18 May 2020 20:09:40 UTC +00:00, points: 1.0, reactable_id: 8, reactable_type: "Article", status: "valid", updated_at: Mon, 18 May 2020 20:09:40 UTC +00:00, user_id: 11>]
```

Reply:
Ohh interesting. Think I was trying to account for that last AR object cache, but I get it now. 👍